### DISCUSSION TITLE
How to setup github [Pull Request Labeler](https://github.com/actions/labeler)

### DISCUSSION THREAD

**Discussion by vimal-verma** [community/community#26851] [Actions]
<p><a href="https://github.com/actions/labeler" rel="noopener nofollow ugc">Pull Request Labeler</a> action showing error</p>
<pre><code class="lang-auto">Error: Error: Input required and not supplied: repo-token
Error: Input required and not supplied: repo-token
</code></pre>
<p>secrets name is LABEL<br>
I have add secrets name in secrets of origination.<br>
what wrong any one please help me</p>

**Comment by airtower-luna** [comment]
<p>It’s hard to say for sure without seeing your workflow, but the message sounds like there’s no <code>repo-token</code> when you call the action. The documentation gives this example:</p>
<pre><code class="lang-yaml">    - uses: actions/labeler@main
      with:
        repo-token: "${{ secrets.GITHUB_TOKEN }}"
</code></pre>
<p>So if you want to use a secret named <code>LABEL</code>, you’d have to do this:</p>
<pre><code class="lang-yaml">    - uses: actions/labeler@main
      with:
        repo-token: "${{ secrets.LABEL }}"
</code></pre>
<p>That <code>LABEL</code> would have to be a PAT with appropriate rights, though in most cases the <code>GITHUB_TOKEN</code> should work.</p>

**Comment by vimal-verma** [comment]
<p>here is <a href="https://github.com/letskhabar/pr-try" rel="noopener nofollow ugc">https://github.com/letskhabar/pr-try</a> Github repo link and <a href="https://github.com/letskhabar/pr-try/tree/master/.github/workflows" rel="noopener nofollow ugc">https://github.com/letskhabar/pr-try/tree/master/.github/workflows</a> is workflow link</p>
<p>thanks</p>

**Comment by airtower-luna** [comment]
<p>Ah, I see the problem. A workflow that’s triggered by a <code>pull_request</code> event from a fork won’t be able to access secrets other than the <code>GITHUB_TOKEN</code>, see <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow">Using encrypted secrets in a workflow</a>.</p>
<p>The first thing I’d try it to check if it works with <code>GITHUB_TOKEN</code>. If not and you really need a PAT you could use the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request_target"><code>pull_request_target</code></a> event instead of <code>pull_request</code>.</p>

**Comment by vimal-verma** [comment]
<p>when I am adding <code>GITHUB_TOKEN</code> secret, it show me a error</p>
<p><code>Failed to add secret. Name is invalid</code></p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You don’t need to create the <code>GITHUB_TOKEN</code>. Github provides it automatically when a workflow is started, I assume that’s why the name isn’t available for your own secret. You just have to <em>use</em> the token, see <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow">Authentication in a workflow</a>. That page actually has an example using <code>actions/labeler</code>, maybe that’ll work for you?</p>
<p>Edit: That page also explicitly states that the <code>GITHUB_TOKEN</code> created for a <code>pull_request</code> event from a fork will only be able to <em>read</em> issues. I assume that’s why <a href="https://github.com/actions/labeler#create-workflow">the example in the <code>actions/labeler</code> Readme</a> also uses <code>pull_request_target</code>.</p>
